doc/iterations/2023.6/shapes/filterTreeBasedRepresentations.adoc

```diff
@@ -0,0 +1,92 @@
+
+=== Scenario 2 - Filter
+
+User want to filter tree items containing a specific value as their label or part of their label.
+User type the shortcut to enable the search bar.
+User type the value to search in the search bar.
+The tree items containing the searched value are highlighted (only the part of the items containing the value).
+User push the filter button (to enable the filter) in the search bar.
+All visible leaf tree items that do not match the search are hidden.
+It will not concern collapsed root/intermediate tree items.
+When expanded, root/intermediate tree items may become hidden if they are now leaf tree items that do not match the search.
+User push again the filter button (to disable the filter) in the search bar.
+All filtered items are not filtered anymore.
+
+=== Scenario 3 - Navigate through results
+
+User want to navigate quickly through the highlighted tree items containing after a search.
+User type the shortcut to enable the search bar.
+User type the value to search in the search bar.
+The tree items containing the searched value are highlighted (only the part of the items containing the value).
+User type the down arrow key to select the next highlighted tree item.
+User type the up arrow key to select the previous highlighted tree item.
+When the last highlighted tree item is selected, the down arrow key has no effect.
+When the first highlighted tree item is selected, the up arrow key has no effect.
+
+=== Cutting backs
+
+Sort the things todo, define the nice-to-have. What could be removed if finally the feature does not fit the appetite.
+1) Search bar open/close
+2) Highlight tree items
+3) Filter tree items
+4) Navigate through highlighted tree items
+5) Display the number of highlighted tree items somewhere in the search bar.
+
+== Rabbit holes
+
+The navigation through up/down arrow keys could lead to some misunderstanding for the users.
+The keys are already used to navigate through the tree items in the Explorer View.
+When the search will be enabled, the up/down arrow keys won't have the same behavior than when the search is disabled.
+An alternate solution could be to have two buttons in the search bar, allowing to navigate through the highlighted tree items.
+
+Some results may not be highlighted because they are collapsed. With the proposed solution, there is no simple way to access to those tree items. This could not be the behavior wanted by some users.
+The documentation will have to explain that.
+
+The filtering could also be confusing for the users. Indeed, the filtering will not concern collapsed root/intermediate tree items. Only leaf tree items are concerned. This could not be the behavior wanted by some users.
+The documentation will have to explain that.
+
+For those two concerns, a solution could be to let users know that there is a match available under a parent item which is collapsed by showing in a tag the number of matches for instance.
```

Comment:
No, this cannot be done with the constraint (frontend only) that have defined this task. This is a major no-go.